### PR TITLE
fix: shade async-profiler packages

### PR DESCRIPTION
**Pull request by zhengziyi0117**
- [ ] Associated with issue https://github.com/apache/skywalking/issues/12748
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Could you post the screenshot of new binary package here?

And, have you tested the feature again? Shading sometimes breaks the function.

**Comment by zhengziyi0117** [comment]
> Could you post the screenshot of new binary package here?你能在这里发布新的二进制包的屏幕截图吗？
> 
> And, have you tested the feature again? Shading sometimes breaks the function.而且，您再次测试过该功能吗？阴影有时会破坏功能。

I have tested it locally and the original function is normal. 

The following is the package structure diagram: 
image

**Comment by wu-sheng** [comment]
About the so files, are they able to move as well?

**Comment by zhengziyi0117** [comment]
> About the so files, are they able to move as well?

The .so file has not been moved yet. 

I think it should not be possible to move the .so file because the async-profiler code gets the dynamic library path based on the following code 
`String resourceName = "/" + getPlatformTag() + "/libasyncProfiler.so";`

**Review by wu-sheng — DISMISSED** [review]

**Review by lujiajing1126 — COMMENTED** [review]
It is a solution as long as you understand that **shading in the agent package instead of agent-core will make it impossible to be used from other packages** like plugins

**Comment by wu-sheng** [comment]
> It is a solution as long as you understand that **shading in the agent package instead of agent-core will make it impossible to be used from other packages** like plugins

Do you mean you prefer to move this(shade) into core module? I think plugins would not access this part of codes.

**Comment by lujiajing1126** [comment]
> > It is a solution as long as you understand that **shading in the agent package instead of agent-core will make it impossible to be used from other packages** like plugins
> 
> Do you mean you prefer to move this(shade) into core module? I think plugins would not access this part of codes.

I am OK with this one.

**Review by lujiajing1126 — APPROVED** [review]

**Comment by wu-sheng** [comment]
@zhengziyi0117 Could you try that?
I think we don't need async profilor accessible from.plugins.

File: apm-sniffer/apm-agent-core/pom.xml [hunk 1]
```diff
@@ -226,6 +228,7 @@
                                     <exclude>org.checkerframework:checker-compat-qual:jar:</exclude>
                                     <exclude>org.checkerframework:checker-qual:jar:</exclude>
                                     <exclude>org.codehaus.mojo:animal-sniffer-annotations:jar:</exclude>
+<!--                                    <exclude>tools.profiler:async-profiler:jar:</exclude>-->
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/pom.xml:231]: I think you should have this?

**Review by wu-sheng — APPROVED** [review]

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**